app/src/features/apiClient/contexts/apiClient.tsx

```diff
@@ -113,14 +118,18 @@ export const ApiClientProvider: React.FC<ApiClientProviderProps> = ({ children }
   const { openTab, deleteTabs, updateTab, replaceTab, updateAddTabBtnCallback } = useTabsLayoutContext();
   const { addNewEnvironment } = useEnvironmentManager();
 
-  const openDraftRequest = useCallback(() => {
-    const requestId = generateDocumentId("apis");
+  const openDraftRequest = useCallback(
+    (callAnalyticEvent: boolean = true) => {
+      const requestId = generateDocumentId("apis");
+      callAnalyticEvent && trackNewTabOpened();
 
-    openTab(requestId, {
-      title: "Untitled request",
-      url: `${PATHS.API_CLIENT.ABSOLUTE}/request/${requestId}?create=true`,
-    });
-  }, [openTab]);
+      openTab(requestId, {
+        title: "Untitled request",
+        url: `${PATHS.API_CLIENT.ABSOLUTE}/request/${requestId}?create=true`,
+      });
+    },
+    [openTab]
+  );
```

Comment:
Instead of having a boolean check for sending event. Can we add the `trackNewTabOpened` to `updateAddTabBtnCallback` so that whenever new tab is clicked it is fired